knowledge-base/pdfprocessing-change-file-size-through-image-quality-and-compression.md

```diff
@@ -0,0 +1,110 @@
+
+```csharp
+    using Telerik.Windows.Documents.Fixed.FormatProviders.Pdf;
+    using Telerik.Windows.Documents.Fixed.FormatProviders.Pdf.Export;
+    using Telerik.Windows.Documents.Fixed.Model;
+
+    //Required in .NET Standard
+    //Telerik.Windows.Documents.Extensibility.JpegImageConverterBase defaultJpegImageConverter = new Telerik.Documents.ImageUtils.JpegImageConverter();
+    //Telerik.Windows.Documents.Extensibility.FixedExtensibilityManager.JpegImageConverter = defaultJpegImageConverter;
+
+    RadFixedDocument document;
+    PdfFormatProvider pdfFormatProvider = new PdfFormatProvider();
+
+    using (Stream input = File.OpenRead("input.pdf"))
+    {
+        document = pdfFormatProvider.Import(input);
+    }
+
+    var settings = new PdfExportSettings();
+
+    #region ImageCompression - None
+    // 182,000 KB file size ||
+    settings.ImageCompression = new ImageFilterTypes[] { ImageFilterTypes.None };
+
+    // ImageQuality has no effect if this is set
+    #endregion
+
+    #region ImageCompression - FlateDecode
+    // 114,000 KB file size || default ImageQuality is High
+    settings.ImageCompression = new ImageFilterTypes[] { ImageFilterTypes.FlateDecode };
+
+    // 114,000 KB file size - no difference with the default setting
+    settings.ImageQuality = ImageQuality.High;
+
+    // 71,000 KB file size
+    settings.ImageQuality = ImageQuality.Medium;
+
+    // 27,000 KB
+    settings.ImageQuality = ImageQuality.Low;
+    #endregion
+
+    #region ImageCompression - Default
+    //around 35,500 KB || default ImageQuality is High
+    settings.ImageCompression = new ImageFilterTypes[] { ImageFilterTypes.Default };
+
+    //around 35,500 KB - no difference with the default setting
+    settings.ImageQuality = ImageQuality.High;
+
+    //around 1,000 KB file size
+    settings.ImageQuality = ImageQuality.Low; 
+
+    //around 3,600 KB file size
+    settings.ImageQuality = ImageQuality.Medium;
+    #endregion
+
+    #region ImageCompression - DCTDecode
+    // 34,500 KB file size || default ImageQuality is High
+    settings.ImageCompression = new ImageFilterTypes[] { ImageFilterTypes.DCTDecode };
+
+    // 34,500 KB file size - no difference with the default setting
+    settings.ImageQuality = ImageQuality.High;
+
+    // 1,000 KB
+    settings.ImageQuality = ImageQuality.Low;
+
+    // 3,600 KB 
+    settings.ImageQuality = ImageQuality.Medium;
+    #endregion
+
+    pdfFormatProvider.ExportSettings = settings;
+
+    using (Stream output = File.OpenWrite("output.pdf"))
+    {
+        pdfFormatProvider.Export(document, output);
+    }
+```
+## See Also
+
+* [PdfProcessing ImageQuality]({%slug radpdfprocessing-concepts-imagequality%})
+* [PdfProcessing ImageCompression](https://docs.telerik.com/devtools/document-processing/libraries/radpdfprocessing/formats-and-conversion/pdf/pdfformatprovider/settings#imagecompression)
```

Comment:
use slug

Reply:
The direct hyperlinks are used to refer to a specific section of the article. ImageCompression is a section of the PdfFormatProvider Settings article.